### PR TITLE
Update performance benchmarks for v2

### PR DESCRIPTION
**Pull request by minrk**
I'm trying to get the benchmarks to run again so we can evaluate our current performance situation. But it's currently failing in most cases.

I've found a few performance things we can improve:

- use single endpoint to check for availability, which is _far_ faster and cheaper than requesting all endpoints
- add concurrency limits to avoid timeouts due to the event-loop being busy rather than the specific action taking a long time (CHP client already has similar for the same reason)

I'm still running into `cannot assign requested address` problems with consul, which appear to be failures in aiohttp? I know that's been an issue with aiohttp in the past, but haven't identified what can be done about it.

placeholder PR for now while I'm making a mess, I'll add a comment when I think this is worth looking at.

### CODE REVIEW

**Comment by minrk** [comment]
This now all works, just breaking up the changes into clearer PRs:

- #165 actually improves the performance
- #166 is some minor cleanup, improving the log output during normal operation
- #167 is actually necessary to get the tests running again at all

**Comment by minrk** [comment]
I re-ran the benchmarks on a big VM, and the important thing is that with the changes here and in traefik v2, we no longer have an O(routes) problem, and performance is consistent and acceptable for up to 500 routes, which was the main thing we were after! #165 was critical for getting the benchmarks to run at all, and setting traefik.providersThrottleDuration to '0s' by default, we see performance much closer to CHP (still, CHP is closer to 1ms while traefik is closer to 100ms, but that's fast enough for us, where before it was closer to 2s because of the throttling)

**Comment by minrk** [comment]
@GeorgianaElena Now that all the actual changes have been merged, this is purely an update to the benchmarks themselves, and a re-run of the data, updating the notebook.

File: performance/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+altair
+jupyter_server
+jupyterhub
+# usually get proxy via `-e .`, but this avoids triggering reinstall every time
+jupyterhub_traefik_proxy
+numpy
```
Comment (manics) [inline performance/requirements.txt:6]:
```suggestion
matplotlib
numpy
```
matplotlib is required by the notebok

**Review by manics — COMMENTED** [review]
I think it'd be helpful to have a short `performance/README.md` explaining where to start (probably just run `<these-scripts>`)

**Comment by minrk** [comment]
@manics good call. Added a README

**Review by GeorgianaElena — APPROVED** [review]
🎉 🎉 🎉

**Pull request merged**